OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -197,13 +199,21 @@ public bool IsCellAcceptable(Actor self, CPos cell)
 			if (!self.World.Map.Contains(cell))
 				return false;
 
+			if (!self.World.ActorMap.GetActorsAt(cell).All(a => a == self))
+				return false;
+
 			if (Info.TerrainTypes.Count == 0)
 				return true;
 
 			var terrainType = self.World.Map.GetTerrainInfo(cell).Type;
 			return Info.TerrainTypes.Contains(terrainType);
 		}
 
+		bool HasAmmo()
+		{
+			return ammoPools.Any(p => p.HasAmmo);
```

Comment:
```suggestion
			return ammoPools.Length == 0 || ammoPools.Any(p => p.HasAmmo);
```

Reply:
Just noticed there's also AmmoUsage, so will need to change it to check for sufficient ammo rather than just any ammo.

Reply:
Good catch(es).

Reply:
The same should be done inside LayMines?

Reply:
Ah, yep.